Rewrite the test_valid_docs test

terminalmage opened the pull request:
This uses a function in the runtests_helpers custom module to perform
all the logic, and only returns what failed the test. This saves us from
having to return the entire contents of sys.doc (as well as log all of
the function calls), and also removes the need to run sys.doc in batches
to get around the "max message size" issue.

Thanks to @s0undt3ch for the idea!

salt-jenkins commented:
@terminalmage, thanks for your PR! By analyzing the history of the files in this pull request, we identified @rallytime, @s0undt3ch and @thatch45 to be potential reviewers.

terminalmage commented:
I've marked this as "Pending Discussion" because I'm not sure it should be merged. It all depends on what the test results show. I can't reproduce this failure locally, so I had to add the debug logging and trigger a jenkins run to test it.

terminalmage commented:
I found and fixed the issue. I've renamed the PR to reflect the fix submitted in this PR.

s0undt3ch reviewed (approved):
Would increasing the batches to 8 buy us more than x weeks?

s0undt3ch commented:
Thinking of this a bit longer, why do we even need to transfer the documentation?
We already have custom modules, let's add a function which checks documentation on the minion side(even better checking the documentation on the master side, a custom runner?) and just returns the problematic modules and any further information required?

terminalmage committed:
Rewrite the tests_valid_docs test

This uses a function in the runtests_helpers custom module to perform
all the logic, and only returns what failed the test. This saves us from
having to return the entire contents of sys.doc (as well as log all of
the function calls), and also removes the need to run sys.doc in batches
to get around the "max message size" issue.

terminalmage commented:
I re-did this PR against 2016.11 and moved the test logic to the runtests_helpers custom module in the test suite.

s0undt3ch reviewed (approved):
Awesome!

terminalmage commented:
Go Go Jenkins!

s0undt3ch commented:
The lint failure is not related.
